### PR TITLE
[INFINITY-1866] Add error messages if Secret is not supported for DC/OS version

### PR DESCRIPTION
**Pull request by mbdcos**
Add Secrets Validator for PodSpec.

### CODE REVIEW

**Comment by mbdcos** [comment]
examples/secrets.yml will generate the following error in DC/OS 1.9 cluster: 
`Caused by: com.mesosphere.sdk.specification.validation.CapabilityValidator$CapabilityValidationException: Dcos-commons (SDK) does not support Secrets for this cluster's DC/OS version
`

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java [hunk 1]
```diff
@@ -68,6 +68,27 @@ private void validate(PodSpec podSpec) throws CapabilityValidationException {
...
             }
         }
+
+        if (!podSpec.getSecrets().isEmpty()) {
+            String fileBAsedSecretSupportMessage = "This cluster's DC/OS version does not support file-based Secrets";
+            // TODO(MB) : Change validator if we decide to support DCOS_DIRECTIVE label
+            // String envBasedSecretSupportMessage =
+            //        "This cluster's DC/OS version does not support environment-based Secrets";
+            String secretsSupportMessage =
+                    "Dcos-commons (SDK) does not support Secrets for this cluster's DC/OS version";
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java]: `This service does not support...` (the reader of this error may not know about dcos-commons or SDK)

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/Capabilities.java [hunk 2]
```diff
@@ -51,6 +51,22 @@ public boolean supportsGpuResource() {
         return hasOrExceedsVersion(1, 9);
     }
 
+    public boolean supportsFileBasedSecrets() {
+        // File-based Secret is supported by DC/OS 1.10 upwards
+        return hasOrExceedsVersion(1, 10);
+    }
+
+    public boolean supportsEnvBasedSecretsProtoBuf() {
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/Capabilities.java]: `Protobuf`

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java [hunk 3]
```diff
@@ -68,6 +68,27 @@ private void validate(PodSpec podSpec) throws CapabilityValidationException {
                 }
             }
         }
+
+        if (!podSpec.getSecrets().isEmpty()) {
+            String fileBAsedSecretSupportMessage = "This cluster's DC/OS version does not support file-based Secrets";
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java]: this is only used within the exception, why not just have it there?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java [hunk 4]
```diff
@@ -68,6 +68,27 @@ private void validate(PodSpec podSpec) throws CapabilityValidationException {
...
+            // TODO(MB) : Change validator if we decide to support DCOS_DIRECTIVE label
+            // String envBasedSecretSupportMessage =
+            //        "This cluster's DC/OS version does not support environment-based Secrets";
+            String secretsSupportMessage =
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java]: this is only used within the exception, why not just have it there?

File: sdk/scheduler/src/test/resources/valid-secrets-env.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    secrets:
+      secret1:
+        secret: hello-world/hello/somePath/secret1
+        env-key: HELLO_SECRET1_ENV
+    tasks:
+      server:
+        goal: RUNNING
+        cmd: "./task-cmd"
+        cpus: 0.1
+        memory: 256
```
Comment (nickbp) [inline sdk/scheduler/src/test/resources/valid-secrets-env.yml]: missing trailing newline

**Comment by mbdcos** [comment]
Manually tested in both 1.9 and 1.10 clusters....

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java (hunk 1 above)
Comment (mbdcos) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java]: This service does not support Secrets for current cluster's DC/OS version

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/Capabilities.java (hunk 2 above)
Comment (mbdcos) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/Capabilities.java]: Protobuf

**Pull request merged**